client/v3/watch_test.go

```diff
@@ -53,3 +55,53 @@ func TestEvent(t *testing.T) {
 		}
 	}
 }
+
+// TestStreamKeyFromCtx tests the streamKeyFromCtx function to ensure it correctly
+// formats metadata as a map[string][]string when extracting metadata from the context.
+//
+// The fmt package in Go guarantees that maps are printed in a consistent order,
+// sorted by the keys. This test verifies that the streamKeyFromCtx function
+// produces the expected formatted string representation of metadata maps when called with
+// various context scenarios.
+func TestStreamKeyFromCtx(t *testing.T) {
+	tests := []struct {
+		md       metadata.MD
+		expected string
+	}{
+		{
+			md: metadata.MD{
+				"key1": []string{"value1"},
+				"key2": []string{"value2a", "value2b"},
+			},
+			expected: "map[key1:[value1] key2:[value2a value2b]]",
+		},
+		{
+			md:       metadata.MD{},
+			expected: "map[]",
+		},
+		{
+			md: metadata.MD{
+				"key1": []string{"value1", "value1a"},
+			},
+			expected: "map[key1:[value1 value1a]]",
+		},
+	}
+
+	// Iterate through each test case and verify the function's output matches
+	// the expected result.
+	for i, tt := range tests {
+		ctx := metadata.NewOutgoingContext(context.Background(), tt.md)
+		actual := streamKeyFromCtx(ctx)
+		if actual != tt.expected {
+			t.Errorf("#%d: streamKeyFromCtx() = %v, want %v", i, actual, tt.expected)
+		}
+	}
```

Comment:
Pls use subtest, 

```
t.Run(tt.name, func(t *testing.T) {
    ......
})
```

For example, 
https://github.com/etcd-io/etcd/blob/4d9dcebd61d63060edd5428c834d01921a03071f/server/etcdserver/api/v3discovery/discovery_test.go#L79-L101